internal/controller/thanosquery_controller_test.go

```diff
@@ -100,22 +99,11 @@ var _ = Describe("ThanosQuery Controller", Ordered, func() {
 				})
 				Expect(err).NotTo(HaveOccurred())
 
-				EventuallyWithOffset(1, func() error {
-					if !utils.VerifyServiceAccountExists(k8sClient, resourceName, ns) {
-						return fmt.Errorf("service account not found")
-					}
-
-					if !utils.VerifyServiceExists(k8sClient, resourceName, ns) {
-						return fmt.Errorf("service not found")
-					}
-
-					if !utils.VerifyDeploymentExists(k8sClient, resourceName, ns) {
-						return fmt.Errorf("deployment not found")
-					}
-
-					return nil
+				EventuallyWithOffset(1, func() bool {
+					return utils.VerifyExistenceOfRequiredNamedResources(
```

Comment:
This is much cleaner! However, we won't what is missing, as it would return false for any of them. We can keep this, but let's log on false output from these functions

Reply:
yep, could be noisy but lets follow up with it

Reply:
Hmm then how about returning some string with false output and only logging that on caller side? Would only log if fails then